### PR TITLE
add namespaces and detached serde

### PR DESCRIPTION
**Pull request by SavinTop**
Alexander, kill me, im ready

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,10 +1,14 @@
 cmake_minimum_required(VERSION 3.0.0)
```
Comment (uselessgoddess) [inline CMakeLists.txt:1]: Is too early version :(

File: src/field_msg.cpp [hunk 2]
```diff
@@ -5,7 +5,9 @@
 #include <bitset>
 #include <ctime>
 
-FieldMessage::FieldMessage()
+namespace msg{
+
+Message::Message()
     : msg_bitmap(0b11),
```
Comment (uselessgoddess) [inline src/field_msg.cpp:11]: fill bitmask zeros to the nearest multiple of eight 
```suggestion
    : msg_bitmap(0b00000011),
```
But it is better to use constants

File: src/field_msg.cpp [hunk 3]
```diff
@@ -21,60 +23,60 @@ FieldMessage::FieldMessage()
     (*this)[e_fields::field_cnt] = {cast(e_fields::field_cnt), field_type::int32, true};
```
Comment (uselessgoddess) [inline src/field_msg.cpp:23]: 🙄

Comment (uselessgoddess) [reply src/field_msg.cpp:23]: Maybe better: 
```cxx
: els({
   {cast(e_fields::field_cnt), field_type::int32, true},
})
```
In fact, I don't know how to improve it. 
I would use a different implementation

File: src/field_msg.cpp [hunk 4]
```diff
@@ -21,60 +23,60 @@ FieldMessage::FieldMessage()
...
 {
     throw std::invalid_argument(str);
 }
 
-bool FieldMessage::is_constant(e_fields fld)
+bool Message::is_constant(e_fields fld)
 {
     return fld == e_fields::message_size ||
            fld == e_fields::corrId ||
            fld == e_fields::field_cnt;
```
Comment (uselessgoddess) [inline src/field_msg.cpp:62]: Sad hard-coding

File: src/field_msg.cpp [hunk 5]
```diff
@@ -21,60 +23,60 @@ FieldMessage::FieldMessage()
...
         _throw("this value cant be modified");
 }
 
-size_t FieldMessage::fld_el::size()
+size_t Message::fld_el::size()
 {
     auto &curr = *this;
     if (curr.type == field_type::int32)
         return sizeof(int32_t);
     return std::get<std::string>(curr.value).size();
```
Comment (uselessgoddess) [inline src/field_msg.cpp:76]: Maybe use `std::visit`??? It's not very pretty, but it's right.

File: src/field_msg.cpp [hunk 6]
```diff
@@ -21,60 +23,60 @@ FieldMessage::FieldMessage()
...
-int FieldMessage::cast(e_fields fld) const
+int Message::cast(e_fields fld)
 {
     return static_cast<int>(fld);
 }
 
-void FieldMessage::_throw(const char *str)
+void Message::_throw(const char *str)
 {
     throw std::invalid_argument(str);
```
Comment (uselessgoddess) [inline src/field_msg.cpp:55]: All your errors is `invalid_argument`?
Maybe create custom type for message errors

File: src/field_msg.cpp [hunk 7]
```diff
@@ -21,60 +23,60 @@ FieldMessage::FieldMessage()
...
     if (curr.type == field_type::int32)
         return sizeof(int32_t);
     return std::get<std::string>(curr.value).size();
 }
 
-void FieldMessage::set(e_fields fld, const std::string &val)
+void Message::set(e_fields fld, const std::string &val)
 {
     throw_on_constant(fld);
     if (val.size() >= UINT16_MAX)
```
Comment (uselessgoddess) [inline src/field_msg.cpp:82]: Maybe create a custom string type whose size cannot be greater than `u16`.

File: src/field_msg.cpp [hunk 8]
```diff
@@ -21,60 +23,60 @@ FieldMessage::FieldMessage()
...
+size_t Message::fld_el::size()
 {
     auto &curr = *this;
     if (curr.type == field_type::int32)
         return sizeof(int32_t);
     return std::get<std::string>(curr.value).size();
 }
 
-void FieldMessage::set(e_fields fld, const std::string &val)
+void Message::set(e_fields fld, const std::string &val)
```
Comment (uselessgoddess) [inline src/field_msg.cpp:79]: Move semantic is your friend
```suggestion
void Message::set(e_fields fld, std::string val)
```

**Review by uselessgoddess — COMMENTED** [review]
bonk

File: src/field_msg.cpp (hunk 3 above)
Comment (SavinTop) [reply src/field_msg.cpp:23]: yeah, me too

**Pull request closed without merging**